service/practitioner_service.go

```diff
@@ -82,9 +82,10 @@ func ValidateAppointmentDetails(svc dao.Service, appointment models.Practitioner
 			log.Info(msg)
 			errs = append(errs, msg)
 		}
-		
+
 		hasValidTransactionID := utils.CheckStringContainsElement(practitioner.Data.Links.Self, "/", transactionID)
-		if !hasValidTransactionID {
+		hasValidAppointmentLinkedToPractitionerID := utils.CheckStringContainsElement(practitioner.Data.Links.Appointment, "/", practitionerID)
```

Comment:
What is the purpose of this check? Should it be combined with the checks for already-appointed practitioner on line 80 or is it for a different purpose?

Reply:
Fixed

Reply:
As earlier discussed, I found the bug I was trying to show while demo to you @nevans4-ch, hence I have reworked the logics as previously done without duplication.